### DISCUSSION TITLE
It is possible to limit one connection per user in order to avoid sharing configuration credentials and setting up multiple clients simultaneously?

### DISCUSSION THREAD

**Discussion by enriluis** [pivpn/pivpn#1783] [Feature requests]
In order to restrict client connections to specific users in OpenVPN and WireGuard, I need to either limit the number of devices that can be connected per user or configure individual connection settings. This is necessary because my users are sharing their credentials with others.

**Comment by orazioedoardo** [comment]
If user A gives their configuration to user B and user B tries to connect while user A is connected, user A will be disconnected and user B will connect. This is the default behavior with OpenVPN unless [duplicate-cn](https://github.com/pivpn/pivpn/blob/701a817fed23e302b91b7677b81c5919fc4ec3f1/files/etc/openvpn/server_config.txt#L36-L38) is used. WireGuard doesn't support using same configuration on multiple clients at the same time at all. If someone attempts to do it, packets will mess up and break the connection all the time.
